build.rs

```diff
@@ -0,0 +1,69 @@
+use primal_sieve::Sieve;
+use std::{fs::File, io::Write, path::Path};
+
+fn checked_naive_factorial(n: u128) -> Option<u128> {
+    let mut acc = 1u128;
+    let mut i = 2u128;
+    while i <= n {
+        acc = acc.checked_mul(i)?;
+        i += 1;
+    }
+    Some(acc)
+}
+
+fn prime_swing(n: u128, sieve: &Sieve) -> Option<u128> {
+    let mut product = 1u128;
+    for prime in sieve.primes_from(2).take_while(|x| *x < n as usize) {
+        let mut p = 1u128;
+        let mut q = n;
+        while q != 0 {
+            q /= prime as u128;
+            // q%2 == 1 if q is odd
+            if q % 2u128 == 1 {
+                p = p.checked_mul(prime as u128)?;
+            }
+        }
+        if p > 1 {
+            product = product.checked_mul(p)?;
+        }
+    }
+    Some(product)
+}
+
+fn main() -> std::io::Result<()> {
+    let path = Path::new("./src/array.rs");
```

Comment:
I'm not sure if writing to the `src` directory (although it will not actually be done) works well once this is packaged. I also think that compiling this file is a bit of a waste of time for everyone installing the crate. Code-gen output should really go into the folder specified by the `OUT_DIR` environment variable. 

How about we move this script into an `examples` folder?

Reply:
I have moved build to an `examples` directory, since its only purpose is to make `array.rs` and it is already done.